torch/linalg/__init__.py

```diff
@@ -8,6 +8,44 @@
 # Note: This not only adds doc strings for functions in the linalg namespace, but
 # also connects the torch.linalg Python namespace to the torch._C._linalg builtins.
 
+cholesky = _add_docstr(_linalg.linalg_cholesky, r"""
+linalg.cholesky(input) -> Tensor
+
+Returns the Cholesky decomposition.
+
+Computes the Cholesky decomposition of a Hermitian (or symmetric for real-valued matrices)
```

Comment:
If these input requirements are not met, do we throw an error? Seems like numpy does:

```
>>> numpy.linalg.cholesky(numpy.random.randn(10, 10))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<__array_function__ internals>", line 6, in cholesky
  File "/home/endoplasm/miniconda3/lib/python3.7/site-packages/numpy/linalg/linalg.py", line 764, in cholesky
    r = gufunc(a, signature=signature, extobj=extobj)
  File "/home/endoplasm/miniconda3/lib/python3.7/site-packages/numpy/linalg/linalg.py", line 91, in _raise_linalgerror_nonposdef
    raise LinAlgError("Matrix is not positive definite")
numpy.linalg.LinAlgError: Matrix is not positive definite
```

I think we should probably have some `self.assertRaisesRegex` tests for any input error cases we expect. Probably would also be good to make sure numpy gives errors for the same cases.

Reply:
Also I think it would be nice to mention in the docs that an error will be thrown if the input requirements are not met.

Reply:
Yes, the error is thrown if the input is not valid, I've added `assertRaisesRegex` checks.
I also added a note to the docs:
> If the :attr:`input` is not Hermitian positive-definite matrix a RuntimeError is raised
          saying that the input is singular and mentioning which minor of the input matrix is not positive-definite.